plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/refactoring/N4JSRefactoringResourceSetProvider.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2019 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ui.refactoring;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.n4js.N4JSGlobals;
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsProvider;
+import org.eclipse.xtext.ui.refactoring.impl.RefactoringResourceSetProvider;
+
+/**
+ * Xtext's RefactoringResourceSetProvider uses LIVE_SCOPE resource loading which does not work for some reason.
+ */
+@SuppressWarnings("restriction")
+public class N4JSRefactoringResourceSetProvider extends RefactoringResourceSetProvider {
+	// public static ResourceSet myGlobalResourceSet;
+
+	@Override
+	public ResourceSet get(IProject project) {
+		if (N4JSGlobals.myGlobalResourceSet != null)
+			return N4JSGlobals.myGlobalResourceSet;
+		// LIVE_SCOPE does not work . PERSISTED_DESCRIPTIONS works. Why?
```

Comment:
mark `TODO`, try asking @szarnekow

Reply:
Thanks for the tip. We are planning to ask @szarnekow :D